es/elastic/sqlalchemy.py

```diff
@@ -29,6 +29,14 @@ class ESDialect(basesqlalchemy.BaseESDialect):
     def dbapi(cls) -> ModuleType:
         return es.elastic
 
+    def has_table(
+        self, connection: Connection, table_name: str, schema: Optional[str] = None, **kwargs: Any
+    ) -> bool:
+        query = "SHOW TABLES"
```

Comment:
This will bring a regression back, since issuing the "special" `SHOW VALID_TABLES` will mix `SHOW TABLES` with an ES cat indices to figure out and exclude empty tables from the result, empty tables don't have columns and that will not play well with SQLAlchemy and Superset

Reply:
There's a PR on Superset for this: https://github.com/apache/superset/pull/16624/files